src/dune_rules/workspace.mli

```diff
@@ -31,6 +31,7 @@ module Context : sig
       ; dynamically_linked_foreign_archives : bool
       ; instrument_with : Lib_name.t list
       ; merlin : bool
+      ; watch_exclusions : string list
```

Comment:
Not sure the `dune-workspace` (equivalently, the `Workspace` module) is the right place for this data, because the Dune file watcher is started at a lower level than that, it is started in the `Scheduler` module (grep for `Dune_file_watcher.create`), where the notion of Workspace does not yet exist. You could put it instead in the Dune `config` file https://dune.readthedocs.io/en/stable/dune-files.html#config (see the `Dune_config` module) and then pass the list of exclusions as an argument to the `Scheduler.go` function and then onwards to `Dune_file_watcher.create`. It is a bit less convenient because it is a global setting, but perhaps OK but a feature that we don't expect to be used that commonly.